ci: disable cirrus cache in 32bit arm job

willcl-ark opened the pull request:
Add an optional matrix field allowing opt-out of configuring cirrus GHA cache when not using cirrus runners.

This is not needed for the cirruslabs/[save|restore]-cache actions, as they automatically fallback based on runner type.

Addresses https://github.com/bitcoin/bitcoin/issues/31965#issuecomment-3252638785

maflcko reviewed (commented):
also, looks like a cache outage in another task?

fanquake commented:
cc @m3dwards

willcl-ark committed:
ci: refactor docker action to return provider str

Avoid relying on future truthy evaluations of string 'false'.

willcl-ark committed:
ci: disable cirrus cache in 32bit arm job

Co-authored-by: Max Edwards <youwontforgetthis@gmail.com>
Add an optional matrix field allowing opt-out of configuring cirrus
GHA cache when not using cirrus runners.

This is not needed for the cirruslabs/[save|restore]-cache actions, as
they automatically fallback based on runner type.

m3dwards commented:
ACK 00c253d494176b31dc4aaba24dc7e61aecb20be2
